goben/prober.go

```diff
@@ -0,0 +1,323 @@
+// Credit to github/google/cloudprober
+// This file is inspired by https://github.com/google/cloudprober/blob/master/probes/ping/ping.go
+// I borrowed the idea of using ICMP (ping) message to measure matrices like rtt. The original
+// implementation is overkilled for our purpose, so this is a simplified version
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"golang.org/x/net/icmp"
+	"golang.org/x/net/ipv4"
+	"log"
+	"math/rand"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	timeBytesSize = 8
+	protocolICMP  = 1
+)
+
+type ProberConfig struct {
+	proto         string        // the protocol for the ICMP packet connection (ie. ip4:icmp, ip4:1, ip6:58 ...)
+	source        string        // the server address
+	target       string      	// the host' addresses
+	probeInterval time.Duration // probe probeInterval
+	pktInterval	  time.Duration // packet sending interval
+	pktsPerProbe  int           // number of packets sent per probe
+	debug		  bool 			// if true, turn on debugging mode => print lots of messages to console
+}
+
+type Prober struct {
+	config		ProberConfig
+	conn		*icmp.PacketConn	// the ICMP connection
+	runCnt		uint64				// a counter that helps to construct seq# and runId
+	result		*csv.Writer			// the csv writer to record measurements
+	file		*os.File			// the csv file descriptor
+}
+
+// Initialize a new Prober instance
+// Need to be called before any of the following
+func (p *Prober) Init(config ProberConfig) error {
+	p.config = config
+
+	// prepare csv writer
+	filePath := fmt.Sprintf("./rtt_%v.csv", config.target)
+	header := []string{"dst", "rtt"}
+	writer, file, err := CreateCSV(filePath, header)
```

Comment:
You create a CSV by default, ideally we only want to create one if explicitly specified. For example, with the --csv command.

Reply:
This is also important so we can configure the filepath where we want to save the file. I think you can just use the schema specified by --csv. You can ignore flowindex for now and always set it to zero since we do only one ping test per direction.